application/src/main/java/org/togetherjava/tjbot/features/chaptgpt/ChatGPTService.java

```diff
@@ -0,0 +1,69 @@
+package org.togetherjava.tjbot.features.chaptgpt;
+
+import com.theokanning.openai.OpenAiHttpException;
+import com.theokanning.openai.completion.chat.ChatCompletionRequest;
+import com.theokanning.openai.completion.chat.ChatMessage;
+import com.theokanning.openai.completion.chat.ChatMessageRole;
+import com.theokanning.openai.service.OpenAiService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Service used to communicate to OpenAI API to generate responses.
+ */
+public class ChatGPTService {
+    private static final Logger logger = LoggerFactory.getLogger(ChatGPTService.class);
+    private static final long TIMEOUT_DURATION = 10000L; // In milliseconds
+    private static final int MAX_TOKENS = 3000;
+    private final OpenAiService openAiService;
+
+    /**
+     * Creates instance of ChatGPTService
+     * 
+     * @param config Config - needed for token to OpenAI API.
+     */
+    public ChatGPTService(Config config) {
+        String token = config.getOpenaiToken();
+        openAiService = new OpenAiService(token, Duration.ofMillis(TIMEOUT_DURATION));
```

Comment:
okay, so here is one change i would like to do. when we want to disable the feature (bad api key), i dont want to constantly send bad requests to the API, but instead disable the feature right here in the code.

Please add some code thats like
```java
if (apiKey.isBlank()) {
  isDisabled = true;
}
```
and when that boolean is set, dont send anything out, but directly respond with empty result.

Reply:
@Taz03 it would be more helpful if u actually give an alternative suggestion instead of just downvoting.

Reply:
its just generally a bad idea to stress an api with loads of requests that u already know will end up failing anyways. worst case, we are IP banned from the api.

if we already know that the user wants to disable chatgpt, we shouldnt contact the api

Reply:
Again - I attempted to resolve this but leaving conversation open versus resolving to get feedback.